.github/workflows/pr-build.yml

```diff
@@ -7,7 +7,30 @@ on:
       - 'main'
 
 jobs:
-  run-pr-checks:
+  changes:
```

Comment:
Minor: can we name this detect-change-scope (or something better if you can think of it). Just so it's obvious what we're doing